### DISCUSSION TITLE
Slideshow feature

### DISCUSSION THREAD

**Discussion by maximizzar** [Radiokot/photoprism-android-client#82] [Ideas]
I would like to ask for the implementation of a slideshow function similar to that of the original web application.  As an additions I'd like to see:

- ability to set an image's preferred showing time
- the ability to randomly cycle through a series of images

**Comment by Radiokot** [comment]
Hi Maximilian, thanks for the idea, will consider adding it.

**Comment by Radiokot** [comment]
Hi there, the feature is added in the version [1.22.0](https://github.com/Radiokot/photoprism-android-client/releases/tag/1.22.0(35))

**Comment by scottrepreneur** [comment]
Should I add another discussion to enhance this slightly to loop the gallery set back to the beginning? The show currently only does one run, I think?

Randomize would be awesome also!

**Reply by ippocratis** [reply]
Randomize and loop are common options in all playlists. 
Makes sense I guess.
+1

**Reply by Radiokot** [reply]
Good idea, will add this 👌🏻

**Reply by Radiokot** [reply]
I've made it looped, making it random is not simple though. It requires preloading all the search results and shuffling them locally, which is resource consuming and takes a long time if your library is big. I don't see adding slideshow randomness in the nearest future.

**Reply by Radiokot** [reply]
[The recent release](https://github.com/Radiokot/photoprism-android-client/releases/tag/1.23.0(36)) has the slideshow looped.
